# Finding genotyped accessions carrying a specific haplotype

!!! example "New feature in v3.7.3"

!!!note

    This section assumes some working knowledge of using the [Genotype tab](/Basic-Functions/Genotype-tab/)

For this specific example we will be using the specific SNP haplotypes for Yr34 used in [user story 2](/User-Stories/User-story-2/).

First navigate to your desired region of interest and load the genotype data for the accessions of interest. In this case we will be using the end of Chr5A is known to contain the Yr34/Yr48 region.

![Image](https://github.com/user-attachments/assets/f8557669-9820-48e8-ab35-2796641fb9ea)

Haplotypes can be selected by clicking on the specific allele located in the ref and alt columns. This will also order the accessions based on the specific selected alleles.

![Image](https://github.com/user-attachments/assets/e428daf7-1997-4c10-9e20-9ee29d148898)

Once the haplotype is selected, open up the sample selection menu and select the "Filter by defined haplotype" option. This will filter the accessions to only those that match the selected haplotype. Select those accessions and add them to the genotype matrix.

![Image](https://github.com/user-attachments/assets/878b91be-ae10-4b67-9881-bbe86fc1deb6)


These results can be then viewed in Prezel or the results can be downloaded as a VCF file via the VCF Download button.

![Image](https://github.com/user-attachments/assets/a5c8474f-33ac-4c55-818e-8c308628436a)


